libs/community/langchain_community/document_loaders/dedoc.py

```diff
@@ -0,0 +1,416 @@
+
+        elif split == "page":
+            nodes = document_tree["content"]["structure"]["subparagraphs"]
+            page_id = nodes[0]["metadata"]["page_id"]
+            page_text = ""
+
+            for node in nodes:
+                if node["metadata"]["page_id"] == page_id:
+                    page_text += self._json2txt(node)
+                else:
+                    yield Document(
+                        page_content=page_text,
+                        metadata={**document_metadata, "page_id": page_id},
+                    )
+                    page_id = node["metadata"]["page_id"]
+                    page_text = self._json2txt(node)
+
+            yield Document(
+                page_content=page_text,
+                metadata={**document_metadata, "page_id": page_id},
+            )
+
+        elif split == "line":
+            for node in document_tree["content"]["structure"]["subparagraphs"]:
+                line_metadata = node["metadata"]
+                yield Document(
+                    page_content=self._json2txt(node),
+                    metadata={**document_metadata, **line_metadata},
+                )
+
+        elif split == "node":
+            yield from self._parse_subparagraphs(
+                document_tree=document_tree["content"]["structure"],
+                document_metadata=document_metadata,
+            )
+
+        else:
+            raise ValueError(
+                f"Got {split} for `split`, but should be one of "
+                f"`{self.valid_split_values}`"
+            )
+
+        if self.with_tables:
+            for table in document_tree["content"]["tables"]:
+                table_text, table_html = self._get_table(table)
+                yield Document(
+                    page_content=table_text,
+                    metadata={
+                        **table["metadata"],
+                        "type": "table",
+                        "text_as_html": table_html,
+                    },
+                )
+
+        for attachment in document_tree["attachments"]:
+            yield from self._split_document(
+                document_tree=attachment,
+                split=self.split,
+                additional_metadata={"type": "attachment"},
+            )
+
+    def _get_table(self, table: dict) -> Tuple[str, str]:
+        """Get text and HTML representation of the table."""
+        table_text = ""
+        for row in table["cells"]:
+            for cell in row:
+                table_text += " ".join(line["text"] for line in cell["lines"])
+                table_text += "\t"
+            table_text += "\n"
+
+        table_html = (
+            '<table border="1" style="border-collapse: collapse; width: 100%;'
+            '">\n<tbody>\n'
+        )
+        for row in table["cells"]:
+            table_html += "<tr>\n"
+            for cell in row:
+                cell_text = "\n".join(line["text"] for line in cell["lines"])
```

Comment:
Could you add HTML escaping here?

Reply:
Do you mean changing `<` to `&lt;`, `>` to `&gt;`, `&` to `&amp;` in `cell_text`?

Reply:
>>> import html
>>> html.escape('<script>malicious</script>')
'&lt;script&gt;malicious&lt;/script&gt;'

Reply:
Done